activestorage/app/jobs/active_storage/attach_remote_file_job.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+require "open-uri"
+
+# Downloads and attaches a File from a URL
+class ActiveStorage::AttachRemoteFileJob < ActiveStorage::BaseJob
+  queue_as { ActiveStorage.queues[:attach_remote_file] }
+
+  discard_on ActiveRecord::RecordNotFound, OpenURI::HTTPError
+
+  def perform(record, name, url)
+    attachment = record.public_send(name)
+    uri = URI.parse(url)
+
+    attachment.attach(io: uri.open, filename: File.basename(uri.path))
```

Comment:
Does `URI#open` timeout? If not, this could be a security risk because malicious actors can specify URL that never returns a response.

Reply:
That's a great point! Is that something that could be configured at the ActiveJob level, or would this solution need to use something like Net::HTTP.start with a `read_timeout`?

Reply:
> Net::HTTP.start with a read_timeout?

This makes more sense because a sensible timeout is the responsibility of the HTTP layer. Is there any case where ActiveJob timeout makes more sense?